pkg/jsontabwriter/jsontabwriter.go

```diff
@@ -0,0 +1,161 @@
+package jsontabwriter
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/ionos-cloud/ionosctl/v6/pkg/constants"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/json2table"
+	"github.com/spf13/viper"
+)
+
+var outputFormatErr = "invalid format: %s"
+
+// TODO: remove cols as function parameter once --cols flag fix is ready
+func GenerateOutput(rootPath string, jsonPaths map[string]string, obj interface{}, cols []string) (string, error) {
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		return generateJSONOutput(obj)
+	}
+
+	if outputFormat == "text" {
+		return generateTextOutput(rootPath, obj, jsonPaths, cols, false)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
+
+func GenerateOutputPreconverted(obj interface{}, convertedObj []map[string]interface{}, cols []string) (string, error) {
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		return generateJSONOutput(obj)
+	}
+
+	if outputFormat == "text" {
+		return generateTextOutput("", convertedObj, nil, cols, true)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
```

Comment:
funcs are redundant - they share too much functionality

we should probably create a new private func, which both of these funcs should use